glslc/src/file_compiler.cc

```diff
@@ -158,6 +188,7 @@ void FileCompiler::SetIndividualCompilationFlag() {
 
 void FileCompiler::SetDisassemblyFlag() {
   if (!preprocess_only_) {
+    output_type_ = OutputType::SpirvAssemblyText;
     disassemble_ = true;
```

Comment:
`disassemble_` and `preprocess_only_` seem redundant now that we have `output_type_`, no?